File: packages/shared-types/execution.ts
import type { IsoDatetimeString } from './json';

export enum ExecutionStatus {
  PENDING = 'pending',
  COMPLETED = 'completed',
  FAILED = 'failed',
}

export interface Execution {
  id: string;
  orderId: string;
  botId: string;
  userId: string;
  filledQuantity: number;
  averagePrice: number;
  fee: number;
  pnl?: number;
  status: ExecutionStatus;
  executedAt: Date;
  createdAt: Date;
}

export interface ExecutionWire {
  id: string;
  order_id: string;
  bot_id: string;
  user_id: string;
  filled_quantity: number;
  average_price: number;
  fee: number;
  pnl?: number;
  status: ExecutionStatus;
  executed_at: IsoDatetimeString;
  created_at: IsoDatetimeString;
}

File: packages/go/domain/execution.go
package domain

import (
	"time"
)

type ExecutionStatus string

const (
	ExecutionStatusPending   ExecutionStatus = "pending"
	ExecutionStatusCompleted ExecutionStatus = "completed"
	ExecutionStatusFailed    ExecutionStatus = "failed"
)

type Execution struct {
	ID             string          `json:"id"`
	OrderID        string          `json:"order_id"`
	BotID          string          `json:"bot_id"`
	UserID         string          `json:"user_id"`
	FilledQuantity float64         `json:"filled_quantity"`
	AveragePrice   float64         `json:"average_price"`
	Fee            float64         `json:"fee"`
	PnL            *float64        `json:"pnl,omitempty"`
	Status         ExecutionStatus `json:"status"`
	ExecutedAt     time.Time       `json:"executed_at"`
	CreatedAt      time.Time       `json:"created_at"`
}
